### PR TITLE
Run fuzzers independently of elm-test

### LINKED ISSUE #82: Split fuzz out into another library
While elm-check exists, ideally, there should be a way to use fuzz without needing to use elm-test. Generating random values in the way that fuzz does is not limited to testing only.

Specifically: we need it for elm-doc-test.

### PR DESCRIPTION
**Pull request by mgold**
This attempts to resolve #82, but ~~⚠️ DO NOT MERGE ⚠️ yet~~ because there is still a bug in either the test code or the real code. My test to shrink a large even number, under the claim that there are no even numbers, should find 0 as the smallest counterexample. Instead it finds 4. There is an error in either my testing code (which is important, that's the reference implementation) or the code under test. It is not a shrinking error, 4 does shrink to 0:

```
> import Lazy.List as LL
> import Shrink
> LL.toList (Shrink.int 4)
[0,2,3] : List Int
```
It's not picking up 2 either...

### CODE REVIEW

**Comment by mgold** [comment]
So it turns out the bug was neither in the library nor the test code but in how shrinking and mapping works. See commit message for the gory details. TL;DR please review @eeue56.

File: src/Test/Runner.elm [hunk 1]
```diff
@@ -147,3 +156,46 @@ formatLabels formatDescription formatTest labels =
                 |> List.map formatDescription
                 |> (::) (formatTest test)
                 |> List.reverse
+
+
+{-| A `Shrinkable a` is an opaque type that allows you to obtain a value of type
+`a` that is smaller than the one you've previously obtained.
+-}
+type Shrinkable a
+    = S { down : LazyList (RoseTree a), over : LazyList (RoseTree a) }
```
Comment (eeue56) [inline src/Test/Runner.elm]: :art: I generally prefer to do this:

```
type alias Shrunk a =
  { down : LazyList (RoseTree a)
  , over : LazyList (RoseTree a)
  }

type Shrinkable a =
  Shrinkable (Shrunk a)
```

That way, it's easier to write functions that dealt with the shrunk type only

Comment (mgold) [inline src/Test/Runner.elm]: I still like passing named arguments explicitly, that's why I'm using the record in the first place, but I got rid of `S` as a name.

**Comment by mgold** [comment]
I'd like to batch this release with a few other changes currently in the works.
